chall01/sadawi.py

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/python3
+
+import sys
+
+format = """.-
+-... 
+-.-. 
+-.. 
+. 
+..-. 
+--. 
+.... 
+.. 
+.--- 
+-.- 
+.-.. 
+-- 
+-. 
+--- 
+.--. 
+--.- 
+.-. 
+... 
+- 
+..- 
+...- 
+.-- 
+-..- 
+-.-- 
+--..""".split()
+
+def error():
+	print("usage: ./sadawi.py <a-zA-Z string>")
+	sys.exit()
+
+if len(sys.argv) != 2:
+	error()
+
+result = ""
+if not sys.argv[1]:
+	error()
+for char in sys.argv[1]:
+	char = char.upper()
+	asciiValue = ord(char)
+	if not 65 <= asciiValue <= 90 and char != " ":
+		error()
+	if char == " ":
+		result += " "
+	else:
+		result += format[asciiValue - 65]
```

Comment:
There is always a so-called "pythonic" way to write code when you do it in Python. I won't describe it in details but here is an example of how it can be done here: 
`result = "".join(format[ord(c) - 65] if c != " " else " " for c in sys.argv[1].upper())`
and if you want to check that every character in the string is alpha, you can do that before (where you check for errors):
```
if not sys.argv[1] or not sys.argv[1].replace(" ", "").isalpha():
    error()
```

Reply:
It's not like you've done something wrong, a lot of experienced programmers don't know where they can do something more "pythonic". But that's one of the main advantages of Python - it's quick to write Python code, and it's laconic. That's why it used so widely in the research field to test a hypothesis, and then, if the hypothesis works, they rewrite it in C++.